### ISSUE TITLE
Bug in the new schema enum default code

### ISSUE THREAD

**Issue by stevewampler** [sksamuel/avro4s#413]
The new schema enum default code is putting the `default` field in the wrong location in the schema.
For example, the `EnumSchemaTest."support default scala enum"` test case generates the following schema where the `default` field is placed outside of the `enum` JSON object:
```
{
  "type": "record",
  "name": "ScalaEnumsWithDefault",
  "namespace": "com.sksamuel.avro4s.schema",
  "fields": [
    {
      "name": "colours",
      "type": {
        "type": "enum",
        "name": "Colours",
        "namespace": "com.sksamuel.avro4s.schema",
        "symbols": [
          "Red",
          "Amber",
          "Green"
        ]
      },
      "default": "Red"
    }
  ]
}
```
According to https://avro.apache.org/docs/current/spec.html#Enums, the `default` field should be at the same level as the `symbols` field:
```
{
  "type": "record",
  "name": "ScalaEnumsWithDefault",
  "namespace": "com.sksamuel.avro4s.schema",
  "fields": [
    {
      "name": "colours",
      "type": {
        "type": "enum",
        "name": "Colours",
        "namespace": "com.sksamuel.avro4s.schema",
        "symbols": [
          "Red",
          "Amber",
          "Green"
        ],
        "default": "Red"
      }
    }
  ]
}
```
Below are two tests that check the compatibility of the above schema when adding a new enum value `Orange`. The first test puts the `default` field in the same location as avro4s 3.0.4 and fails the compatibility test. The second test puts the `default` field next to the `symbols` field and succeeds:
```import com.sksamuel.avro4s.AvroSchema
import org.apache.avro.SchemaCompatibility
import org.apache.avro.SchemaCompatibility.SchemaCompatibilityType
import org.junit.jupiter.api.Assertions._
import org.junit.jupiter.api.Test

class Avro4sEnumDefaultTest {
  object Colours extends Enumeration {
    val Red, Amber, Green = Value
  }

  case class ScalaEnumsWithDefault(colours: Colours.Value = Colours.Red)

  // This test uses the AVRO SchemaCompatibility.checkReaderWriterCompatibility method to check the compatibility of two
  // versions of an enum schema with a default enum field in the location produced by avro4s 3.0.4.
  // In the first version of the schema, the
  // enum values are "Red", "Amber", and "Green". In the second version, "Orange" is added to the enum values.
  //
  // The test shows that the two schemas versions are not actually compatible because the "default" enum value field
  // ("default": "Red") is output in the wrong location in the schema.
  @Test
  def avro4sDefaultEnumTest1: Unit = {
    // this is the schema generated by avro4s 3.0.4
    val schemaVersion1 = new org.apache.avro.Schema.Parser().parse(
      """
        |{
        |  "type": "record",
        |  "name": "ScalaEnumsWithDefault",
        |  "namespace": "Avro4sEnumDefaultTest",
        |  "fields": [
        |    {
        |      "name": "colours",
        |      "type": {
        |        "type": "enum",
        |        "name": "Colours",
        |        "symbols": [
        |          "Red",
        |          "Amber",
        |          "Green"
        |        ]
        |      },
        |      "default": "Red"
        |    }
        |  ]
        |}""".stripMargin)

    // just to be sure, actually generate the schema and check against the one above
    val avro4sSchema = AvroSchema[ScalaEnumsWithDefault]

    assertEquals(avro4sSchema, schemaVersion1)

    // this schema adds a new enum value "Orange". It should be compatible the the prior version of the schema
    // due to the default value, but it's not because the "default" value is in the wrong location.
    val schemaVersion2 = new org.apache.avro.Schema.Parser().parse(
      """
        |{
        |  "type": "record",
        |  "name": "ScalaEnumsWithDefault",
        |  "namespace": "Avro4sEnumDefaultTest",
        |  "fields": [
        |    {
        |      "name": "colours",
        |      "type": {
        |        "type": "enum",
        |        "name": "Colours",
        |        "symbols": [
        |          "Red",
        |          "Amber",
        |          "Green",
        |          "Orange"
        |        ]
        |      },
        |      "default": "Red"
        |    }
        |  ]
        |}""".stripMargin)

    val schemaPairCompatibility: SchemaCompatibility.SchemaPairCompatibility = SchemaCompatibility.checkReaderWriterCompatibility(
      schemaVersion1,
      schemaVersion2
    )

    // this assert fails
    assertEquals(SchemaCompatibilityType.COMPATIBLE, schemaPairCompatibility.getType)
  }

  // This test uses the AVRO SchemaCompatibility.checkReaderWriterCompatibility method to check the compatibility of two
  // versions of an enum schema with a default enum field placed next to the "symbols" field as specified in
  // https://avro.apache.org/docs/current/spec.html#Enums.
  //
  // In the first version of the schema, the
  // enum values are "Red", "Amber", and "Green". In the second version, "Orange" is added to the enum values.
  //
  // The test shows that the two schemas versions are compatible.
  @Test
  def avro4sDefaultEnumTest2: Unit = {
    // this is the schema generated by avro4s 3.0.4
    val schema1WithDefaultOutsideEnumType = new org.apache.avro.Schema.Parser().parse(
      """
        |{
        |  "type": "record",
        |  "name": "ScalaEnumsWithDefault",
        |  "namespace": "Avro4sEnumDefaultTest",
        |  "fields": [
        |    {
        |      "name": "colours",
        |      "type": {
        |        "type": "enum",
        |        "name": "Colours",
        |        "symbols": [
        |          "Red",
        |          "Amber",
        |          "Green"
        |        ],
        |        "default": "Red"
        |      }
        |    }
        |  ]
        |}""".stripMargin)

    // this schema adds a new enum value "Orange". It should be compatible the the prior version of the schema
    // due to the default value.
    val schema1ModifiedWithDefaultOutsideEnumType = new org.apache.avro.Schema.Parser().parse(
      """
        |{
        |  "type": "record",
        |  "name": "ScalaEnumsWithDefault",
        |  "namespace": "Avro4sEnumDefaultTest",
        |  "fields": [
        |    {
        |      "name": "colours",
        |      "type": {
        |        "type": "enum",
        |        "name": "Colours",
        |        "symbols": [
        |          "Red",
        |          "Amber",
        |          "Green",
        |          "Orange"
        |        ],
        |        "default": "Red"
        |      }
        |    }
        |  ]
        |}""".stripMargin)

    val schemaPairCompatibility: SchemaCompatibility.SchemaPairCompatibility = SchemaCompatibility.checkReaderWriterCompatibility(
      schema1WithDefaultOutsideEnumType,
      schema1ModifiedWithDefaultOutsideEnumType
    )

    // succeeds
    assertEquals(SchemaCompatibilityType.COMPATIBLE, schemaPairCompatibility.getType)
  }
}

**Comment by stevewampler** [comment]
@sksamuel I'm wondering what's involved to create another release of avro4s to include the merged fix?

Also, should I close this issue now, or wait for a release?

**Comment by sksamuel** [comment]
It looks like the 3.0.5 I did went missing. I've just released 3.0.6.

**Comment by stevewampler** [comment]
Thanks @sksamuel !
